core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt

```diff
@@ -0,0 +1,25 @@
+package com.wachon.spotiwrap.data.repository
+
+import com.wachon.spotiwrap.core.common.model.ArtistModel
+import com.wachon.spotiwrap.core.network.datasource.NetworkSpotifyDatasource
+import com.wachon.spotiwrap.data.extensions.toArtistModel
+import com.wachon.spotiwrap.data.worker.Syncable
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+interface SearchRepository : Syncable {
+    suspend fun searchArtist(query: String): Flow<List<ArtistModel>>
+}
+
+class DefaultSearchRepository(
```

Comment:
No need of this. UserRepo and ArtistRepo already exists

Reply:
+1

Reply:
Done, I have already split the methods between Playlist Repo and Artist Repo